File: youlexuan_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;

/**
 * 品牌服务层接口
 */
public interface BrandService {

    /**
     * 返回全部列表
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);

    /**
     * 添加品牌
     */
    public void add(TbBrand brand);

    /**
     * 根据id获取信息
     */
    public TbBrand findOne(Long id);
    /**
     * 修改品牌
     */
    public void update(TbBrand brand);

    /**
     * 删除
     */
    public void delete(Long[] ids);
}

File: youlexuan_manager_web/src/main/webapp/js/service/brandService.js
//定义service层
app.service('brandService',function ($http) {
    //读取列表数据
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    };
   /* //获取分页数据
    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    };*/
    //增加品牌
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    };
    //回显
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };
    //修改
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    };
    //删除
    this.delete=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    };
    //获取列表数据 搜索功能
    this.seach=function (page, rows, searchEntity) {
        return $http.post('../brand/findPage.do?page='+page+'&rows='+rows,searchEntity);
    }
});